core/analysis.go

```diff
@@ -528,3 +531,45 @@ func AnalyzeRepositories(sess *Session) {
 	wg.Wait()
 
 }
+
+// WriteOutput will write a csv or json file to disk based on user specified options containing discovered findings
+func WriteOutput(sess *Session) {
+	if len(sess.Findings) > 0 {
+		if len(sess.OutputFile) != 0 {
+			f, err := os.Create(sess.OutputFile)
+			if err != nil {
+				sess.Out.Error("Failed writing to csv with error:\n%s\n", err)
+			}
+			defer f.Close()
+			if sess.CSV == true {
+				sess.Out.Info("Writing results in CSV format to: %s\n", sess.OutputFile)
+				w := csv.NewWriter(f)
+				defer w.Flush()
+
+				fields := sess.Findings[0].getFieldNames()
+				w.Write(fields)
+				for _, v := range sess.Findings {
+					_ = w.Write(v.getValues())
+				}
+				w.Flush()
+				f.Close()
+			} else if sess.JSON == true {
```

Comment:
This is great here, but we should be thinking a little ahead and by the time we get to the third **else if** we should think about using a switch instead if possible. Thinking at scale early on is far better than retro-fitting after the fact. Not saying it is necessary here, but it is something to consider as we look at various tools such as this.

Again, I stress, the code is great and very much appreciated, I just feel we can make sure we keep the bar high when compared to other tools like this. This is a fairly crowded field and my goal it to be near the top of the pack when this goes stable.
@funkwhatyouheard

Reply:
Considered a switch statement instead from the jump but didn't see a lot of benefit at this stage. Totally fine with changing it to a switch now with future development in mind.

Reply:
Thinking more on this, I don't know that the current implementation is a good candidate for a switch b/c it cues off two different variables. We'd have to make something like a --output-format param that takes a list of comma separated strings like csv, json, xml, etc. Think it's worth changing? Not a major tweak but want to make sure it's worthwhile before pursuing.